### PR TITLE
[WIP] Rust 2018 Transition

### PR DESCRIPTION
**Pull request by dbcfd**
Update rust code to be compatible with Rust 2015 or 2018. This requires rust 1.30.0+.

More information on which OS's support which rust versions can be found [here](https://redmine.openinfosecfoundation.org/issues/2507)

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/2629

Describe changes:
- Run `cargo fix --edition` to make compatible with 1.30.0+ rust
- Run `cargo fmt` to format

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

**Comment by jasonish** [comment]
> @jasonish how do you feel about the cargo fmt run? I see the value of using it, but I dislike the big reformatting commit.

I have two thoughts on the matter.  If no one is actively working on branches, go for the system wide format, and then we make it part of PR review for the future.  This gets it done with minimal affect to others.

If there are active branches, then I'd lean more towards all PR's doing rustfmt just on the files they touch. Then we'll get eventually consistent with respect to formatting.

I have nothing outstanding that wouldn't be easy to merge.  If there was a time to do a wide merge, it would probably be now (well as soon as the 5.0 branch "opens").

**Comment by chifflier** [comment]
Hi,
FYI, cargo fix will not add more compatibilty (code already builds with 1.32), but breaks 1.24.1 (notation `use crate::mod` is unsupported).
So, I would advice **not** to merge this at this point, if 1.24.1 is still a target.
This will be addressed in the nom4 transition, the upcoming PR is compatible with both 1.24.1 and stable (currently 1.32)

**Comment by dbcfd** [comment]
I'm on the fence with regards to cargo fmt. Unless we actually pull it in as part of the build process (or manually check PRs), it is likely that code will not remain formatted, since developers may forget to run cargo fmt.

I do like it for the uniformity it provides, especially given the uniformity that the c code has. This makes readability and maintenance much easier.

I'm going to mark this as WIP, to indicate it is not meant to be merged. It is mainly meant to determine how we want to proceed with cargo fix and cargo fmt, or if we even want to use them.

File: rust/src/applayertemplate/logger.rs [hunk 1]
```diff
@@ -15,10 +15,10 @@
  * 02110-1301, USA.
  */
 
+use super::template::TemplateTransaction;
+use crate::json::*;
```
Comment (chifflier) [inline rust/src/applayertemplate/logger.rs]: For ex, this line breaks with rust 1.24.1:
```
error: `crate` in paths is experimental (see issue #45477)
  --> src/applayertemplate/logger.rs:19:5                                                                 
   |                                                                                          
19 | use crate::json::*;                                                  
   |     ^^^^^           
```

**Comment by dbcfd** [comment]
Closing. Think the consensus is we could use cargo fix when appropriate rust versions are available on supported OS's, but cargo fmt is too disruptive for minimal value.

**Pull request closed without merging**